add gpg content (#253)
githubtraining/training-manual: docs/24_commit_sig_verification.md

## Commit Signature Verification

Verifying the authenticity of the source of commits submitted to your repository might be important to you and your team. GitHub supports methods of ensuring that the author of the commit in a pull request was actually created from a trusted source. Commit signature verification is a cryptographically verifiable method of ensuring that commits coming from Mona Lisa are in fact coming from Mona Lisa.

> Signed commits can be created by implementing GPG or S/MIME signature that is attached to every commit that you make. Once you have integrated GPG or S/MIME on your machine and account, commits you create will appear like this in the GitHub UI:

Reviewer: ```suggestion
Signed commits can be created by implementing GPG or S/MIME signature that is attached to every commit that you make on your local machine (downstream repository). Once you have integrated GPG or S/MIME on your machine and account, commits you create locally and push to the upstream will appear like this in the GitHub UI:
```

![Signed commit example](https://docs.github.com/assets/images/help/commits/verified-commit.png)

> An unverified commit will not display the "Verified" tag in the GitHub interface.

Reviewer: ```suggestion
An unsigned commit on the local downstream repo will not display the "Verified" tag in the GitHub interface. Any commits made directly on the GitHub interface will automatically include the "Verified" tag.
```

1. Enter the following command to list the GPG keys that have both a public and private key:

    ```bash
    gpg --list-secret-keys --keyid-format LONG
    ```

1. From the list of GPG keys, copy the GPG key ID you would like to use.
1. Enter the following command to set the GPG signing key you want to use in Git, replace `GPG_key_ID` with the value of the GPG key you just copied:

    ```bash
    git config --global user.signingkey GPG_key_ID
    ```

### Activity: Signing commits with your signing key

1. On your computer, access the `github-games` repository you cloned before.
1. Create a new branch named `add-author` by entering:

    ```bash
    git checkout master
    git checkout -b add-author
    ```

1. Open the `README.md` file in your favorite text editor.
1. On the last line of the `README.md` file, potentially line 10, add the following, replacing `@username` with your GitHub username:

    ```bash
    Modified during a training by @username
    ```

1. Create a commit and push your changes by entering the following:

    ```bash
    git add README.md
    git commit -S -m  "Add author to README"
    git push -u origin add-author
    ```

> 1. Create a pull request and in the **Commits** tab you will see the **Verified** tag next to your commit.

Reviewer: ```suggestion
1. Create a pull request and in the **Commits** tab you will see the **Verified** tag next to your commit.

> Note: If you've enforced branch protections in your repository, you can set up required commit signing. When you enable required commit signing on a branch, contributors can only push local commits to the branch if they are signed and verified. Contributors can merge signed and verified commits using the GitHub Enterprise web interface. For more information, see [Configuring protected branches.](https://docs.github.com/en/enterprise/2.15/user/articles/configuring-protected-branches)
```
